src/command_modules/azure-cli-sf/tests/test_sf_commands.py

```diff
@@ -0,0 +1,154 @@
+import os
+
+from azure.cli.testsdk import ScenarioTest, JMESPathCheck, \
+JMESPathCheckExists, NoneCheck
+from azure.cli.testsdk.base import execute
+from azure.cli.testsdk.preparers import AbstractPreparer
+
+
+class SelectNoSecClusterPreparer(AbstractPreparer):
+    def __init__(self, parameter_name="endpoint",
+                 endpoint="http://127.0.0.1:10550",
+                 env_variable_name="AZURE_CLI_SF_ENDPOINT"):
+        # Name randomization unnecessary
+        super(SelectNoSecClusterPreparer, self).__init__("test", 10)
+        self.endpoint = endpoint
+        self.parameter_name = parameter_name
+        self.env_variable_name = env_variable_name
+
+    def create_resource(self, _, **kwargs):
+        # Omit name here since there is no randomization required
+        endpoint = os.environ.get(self.env_variable_name, self.endpoint)
```

Comment:
So unfortunately, unlike mgmt command modules our tests require that whoever makes the change provisions a Service Fabric cluster, and then runs this small set of tests. Hence, the use of the environment variable since right now we cannot provision a cluster from the CLI for every CI test, it would take far too long.

We could make these manual tests and omit them from CI, telling users if they want to modify the `sf` commands, they need to provision a cluster then run these tests.

I'm not sure what's best here, unfortunately we don't have a good light weight testing mechanism since a these commands end up exercising long code paths in the Service Fabric clusters.